.buildkite/pipeline.yml

```
@@ -1,3 +1,6 @@
 steps:
-  - label: Validate the plugin
-    command: make validate
+  - command: make validate
+  # node v18 and later doesn't work on the current build agent.
+  # Once we switch to a newer OS for the default agent, we can add more node
+  # version here. See https://github.com/nvm-sh/nvm/issues/2972
+  - command: .buildkite/dogfood-pipeline.sh v17
```

Instead of generating the pipeline using a shell script and have an additional level of indirection, what about using a `matrix` like this directly in the `pipeline.yml` (and get rid of `dogfood-pipeline.sh` and its generated `dogfood-pipeline.yml`)?

```suggestion
  # node v18 and later doesn't work on the current build agent.
  # Once we switch to a newer OS for the default agent, we can add more
  # node versions in the matrix. See https://github.com/nvm-sh/nvm/issues/2972
  - label: ':node: Validate node {{matrix}}'
    command: .buildkite/verify_node.sh '{{matrix}}'
    plugins:
      - automattic/nvm#${BUILDKITE_COMMIT}:
          version: '{{matrix}}'
    matrix:
      - v17
```

---

Neat! I didn't realise you can use matrix on plugin configurations too.